core/domain/platform_parameter_registry_test.py

```diff
@@ -247,92 +248,49 @@ def test_update_parameter_with_invalid_rules_failure(self) -> None:
         self
     ) -> None:
         parameter_name = 'parameter_a'
         registry.Registry.create_feature_flag(
             ParamNames.PARAMETER_A, 'dev feature', FeatureStages.DEV)
 
-        with self.assertRaisesRegex(
-            utils.ValidationError,
-            'Feature in dev stage cannot be enabled in test or production '
-            'environments.'):
-            registry.Registry.update_platform_parameter(
-                parameter_name,
-                feconf.SYSTEM_COMMITTER_ID,
-                'commit message',
-                [
-                    parameter_domain.PlatformParameterRule.from_dict({
-                        'filters': [
-                            {
-                                'type': 'server_mode',
-                                'conditions': [['=', FeatureStages.TEST.value]]
-                            }
-                        ],
-                        'value_when_matched': True
-                    })
-                ],
-                False
-            )
-
-    def test_update_dev_feature_with_rule_enabled_for_prod_raises_exception(
+        with self.swap(constants, 'DEV_MODE', False):
+            with self.swap(feconf, 'ENV_IS_OPPIA_ORG_PRODUCTION_SERVER', True):
+                with self.assertRaisesRegex(
+                    utils.ValidationError,
+                    'Feature in dev stage cannot be enabled in prod '
+                    'environment.'
+                ):
+                    registry.Registry.update_platform_parameter(
+                        parameter_name,
+                        feconf.SYSTEM_COMMITTER_ID,
+                        'commit message',
+                        [],
+                        True
+                    )
+
+    def test_update_dev_feature_in_test_environment_raises_exception(
         self
     ) -> None:
         parameter_name = 'parameter_a'
         registry.Registry.create_feature_flag(
             ParamNames.PARAMETER_A, 'dev feature', FeatureStages.DEV)
 
-        with self.assertRaisesRegex(
-            utils.ValidationError,
-            'Feature in dev stage cannot be enabled in test or production '
-            'environments.'):
-            registry.Registry.update_platform_parameter(
-                parameter_name,
-                feconf.SYSTEM_COMMITTER_ID,
-                'commit message',
-                [
-                    parameter_domain.PlatformParameterRule.from_dict({
-                        'filters': [
-                            {
-                                'type': 'server_mode',
-                                'conditions': [['=', FeatureStages.PROD.value]]
-                            }
-                        ],
-                        'value_when_matched': True
-                    })
-                ],
-                False
-            )
-
-    def test_update_test_feature_with_rule_enabled_for_prod_raises_exception(
```

Comment:
The corresponding test for this seems missing.

Reply:
I have added it.